dask/dataframe/tests/test_dataframe.py

```diff
@@ -3615,6 +3615,38 @@ def test_dataframe_reductions_arithmetic(reduction):
     )
 
 
+@pytest.mark.filterwarnings()
+def test_dataframe_mode():
+    data = [["Tom", 10, 7], ["Farahn", 14, 7], ["Julie", 14, 5], ["Nick", 10, 10]]
+
+    df = pd.DataFrame(data, columns=["Name", "Age", "Num"])
+    ddf = dd.from_pandas(df, npartitions=3)
+
+    # test series with object dtype
+    assert_eq(
+        ddf.Name.mode().compute().sort_values().reset_index(drop=True), (df.Name.mode())
+    )
```

Comment:
This mode implementation doesn't sort the returned values as pandas does so when I compare the results I sort_values and reset_index first.

Reply:
Ideally we would have the following:

```python
assert_eq(ddf.Name.mode(), df.Name.mode())
assert_eq(ddf.mode(), df.mode())
```

Yes, I agree with your recent commend that it should be fine to call operations like sorting and reset index on the single partition result value.

It is important to not call `.compute` in `assert_eq`.  This lets us run a variety of important tests on the Dask object.

Reply:
@mrocklin I've removed the .compute() in testing locally and found an issue which I hadn't noticed before.

The issue is when computing mode for each row of a dataframe (`df.mode(axis=1)`), the number of columns returned isn't known until after mode is computed. (see image)
![image](https://user-images.githubusercontent.com/23342526/77353484-0be95700-6d0f-11ea-9cf9-f51e617ff508.png)

This prevents me from specifying metadata in the typical way when axis=1.  Is there a way to handle a variable number of columns in dask dataframe metadata?  If not, I think I'll leave the axis parameter unimplemented in the mode method for dask's dataframes.  Does that seem reasonable?

Reply:
I removed the `axis` parameter from mode and updated tests to not call compute.